### PR TITLE
Charlemagne mid level api

### PR DESCRIPTION
**Pull request by KyleGoyette**
This PR adds a mid-level API over the low-level api (_add_panel) that allows users to create a custom plot on a table.
It also allows the plot_table function to be called as wandb.plot_table

### CODE REVIEW

File: wandb/sdk/wandb_run.py [hunk 1]
```diff
@@ -567,6 +573,23 @@ def _history_callback(self, row=None, step=None):
...
                 row[k] = row[k].value
                 visualize_persist_config = True
+            elif isinstance(row[k], CustomChart):
+                custom_charts[k] = row[k]
+                custom_chart = row[k]
+
+        for k, custom_chart in custom_charts.items():
+            # remove the chart key from the row
+            # TODO: is this really the right move? what if the user logs
+            #     a non-custom chart to this key?
```
Comment (KyleGoyette) [inline wandb/sdk/wandb_run.py:583]: If I understand the flow correctly, this should only occur when the last key logged to k is a custom chart, so it should only ever happen if the key being removed is a custom chart.

File: wandb/sdk/wandb_run.py [hunk 2]
```diff
@@ -923,6 +946,23 @@ def finish(self, exit_code=None):
     def join(self, exit_code=None):
         self.finish(exit_code=exit_code)
 
+    def plot_table(self, vega_spec_name, data_table, fields, string_fields=None):
+        """Creates a custom plot on a table.
+        Args:
+            vega_spec_name: the name of the spec for the plot
+            table_key: the key used to log the data table
```
Comment (Xyzrr) [inline wandb/sdk/wandb_run.py:953]: is the table_key arg outdated?

**Review by shawnlewis — APPROVED** [review]

File: wandb/sdk/wandb_run.py (hunk 2 above)
Comment (KyleGoyette) [reply wandb/sdk/wandb_run.py:953]: I'm going to update the design doc to reflect the final status of this.

File: wandb/sdk/wandb_run.py [hunk 3]
```diff
@@ -1456,7 +1496,7 @@ def _show_history(self):
         else:
             wandb.termlog("Run history:")
             history_lines = ""
-            format_str = u"  {:>%s} {}\n" % max_len
+            format_str = "  {:>%s} {}\n" % max_len
```
Comment (raubitsj) [inline wandb/sdk/wandb_run.py:1499]: why do you want to break python2?

File: wandb/plots/explain_text.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import wandb
 from wandb import util
-from wandb.plots.utils import test_missing, test_types, encode_labels
+from wandb.plots.utils import test_missing, test_types, encode_labels, deprecation_notice
```
Comment (raubitsj) [inline wandb/plots/explain_text.py:3]: remove deprecation stuff, lets deprecate after both interfaces exist for a few releases

**Pull request merged**